elements/bulbs-video/elements/carousel.js

```diff
@@ -0,0 +1,81 @@
+import { BulbsHTMLElement, registerElement } from 'bulbs-elements/register';
+
+import './carousel.scss';
+
+export default class Carousel extends BulbsHTMLElement {
+  getAnchors () {
+    return this.querySelectorAll('bulbs-video-carousel-item a');
+  }
+
+  getActiveCarouselItem () {
+    return this.querySelector(
+      `bulbs-video-carousel-item[href='${window.location.pathname}']`
+    );
+  }
+
+  createdCallback () {
+    this.playerEnded = this.playerEnded.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+
+    this.videoPlayer = this.querySelector('bulbs-video');
+    this.slider = this.querySelector('bulbs-video-carousel-slider');
+    this.addEventListener('click', this.handleClick);
+  }
+
+  attachedCallback () {
+    if (this.videoPlayer) {
+      // This event MUST pass false in as the third argument,
+      //  video events do not bubble, so we have to listen to the
+      //  catpure phase.
+      this.videoPlayer.addEventListener('ended', this.playerEnded, true);
```

Comment:
We should try to make a note of these event listeners of the player dependency so we remember to change it with the JWPlayer implementation.